packages/client-sdk-web/src/internal/auth-client.ts

```diff
@@ -0,0 +1,81 @@
+import {auth} from '@gomomento/generated-types-webtext';
+import {CredentialProvider, MomentoLogger, GenerateApiToken} from '..';
+import {version} from '../../package.json';
+import {Configuration} from '../config/configuration';
+import {Request, UnaryInterceptor, UnaryResponse} from 'grpc-web';
+import {Header, HeaderInterceptorProvider} from './grpc/headers-interceptor';
+import {_GenerateApiTokenRequest} from '@gomomento/generated-types-webtext/dist/auth_pb';
+import {cacheServiceErrorMapper} from '../errors/cache-service-error-mapper';
+import {_GenerateApiTokenResponse} from '@gomomento/common';
+import Never = _GenerateApiTokenRequest.Never;
+import Expires = _GenerateApiTokenRequest.Expires;
+
+export interface AuthClientProps {
+  configuration: Configuration;
+  credentialProvider: CredentialProvider;
```

Comment:
Do we want a credentials provider as props? This means that this client would require an auth token. I thought we wanted to be able to initialize it without a token

Reply:
Good call.

Reply:
@bruuuuuuuce hmmm but I still need an auth token to get a control endpoint.
So we don't need to pass an auth token to the auth header when calling the generate api token api, but I still need an auth token to get a control endpoint, right?

Reply:
Ah, never mind. We have our own end pointer creater.